htmlproofer update in kiali.io

htmlproofer has been updated to V. 4.2.0 (It was v. htmlproofer 3.19.3 in my local environment) and the parameters have been changed, causing an OptionParser error: 

```
Total in 1651 ms
htmlproofer 4.2.0 | Error:  Whoops, we can't understand your command.
htmlproofer 4.2.0 | Error:  invalid option: --empty_alt_ignore
htmlproofer 4.2.0 | Error:  Run your command again with the --help switch to see available options.
Traceback (most recent call last):
	4: from /usr/local/bin/htmlproofer:23:in `<main>'
	3: from /usr/local/bin/htmlproofer:23:in `load'
	2: from /usr/local/share/gems/gems/html-proofer-4.2.0/bin/htmlproofer:11:in `<top (required)>'
	1: from /usr/local/share/gems/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
/usr/local/share/gems/gems/mercenary-0.4.0/lib/mercenary/program.rb:33:in `go': invalid option: --empty_alt_ignore (OptionParser::InvalidOption)
make: *** [Makefile:74: validate-site] Error 1
```
I've found this error in https://github.com/kiali/kiali.io/pull/537
I was able to reproduce it locally without the use of the cache (--no-cache) and force the image generation: 

make -e FORCE_BUILD=true validate-site

It looks like the parameters have been changed, and might be also have change the validation rules. 

- https://github.com/gjtorikian/html-proofer
- https://github.com/gjtorikian/html-proofer/blob/main/CHANGELOG.md

Remove html-proofer version from Dockerfile (Revert https://github.com/kiali/kiali.io/pull/559)

--- a/Dockerfile
+++ b/Dockerfile
@@ -26,7 +26,7 @@ RUN tar xzvf /hugo.tar.gz -C / \
 RUN npm -g -D install postcss postcss-cli autoprefixer
 
 # Install html-proofer
-RUN gem install html-proofer -v 3.19.4
+RUN gem install html-proofer
 
 # Confirm htmlproofer binary is available and show its version
 RUN htmlproofer --version

--- a/Makefile
+++ b/Makefile
@@ -72,7 +72,7 @@ URL_IGNORE:=$(URL_IGNORE)$(NEW_URLS)
 CONFIG='{\"enforce_https\": false,\"check_internal_hash\": false, \"check_external_hash\":false,\"ignore_empty_alt\":true}'
 .PHONY: validate-site
 validate-site: build-hugo
-	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --assume-extension --enforce_https --check_internal_hash --check_external_hash --ignore_empty_alt --ignore_urls \"${URL_IGNORE}\" ./public"
+	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume-extension --allow-missing-href --ignore_missing_alt --check-external-hash --check_internal_hash --enforce-https false --ignore_urls \"${URL_IGNORE}\""
 	#${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --assume-extension --enforce_https --check_internal_hash --check_external_hash --ignore_empty_alt --log_level=debug --ignore_urls \"${URL_IGNORE}\" ./public"
 	#${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --swap_attributes ${CONFIG} ./public"
 

--- a/Makefile
+++ b/Makefile
@@ -72,7 +72,5 @@ URL_IGNORE:=$(URL_IGNORE)$(NEW_URLS)
 CONFIG='{\"enforce_https\": false,\"check_internal_hash\": false, \"check_external_hash\":false,\"ignore_empty_alt\":true}'
 .PHONY: validate-site
 validate-site: build-hugo
-	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume-extension --allow-missing-href --ignore_missing_alt --check-external-hash --check_internal_hash --enforce-https false --ignore_urls \"${URL_IGNORE}\""
-	#${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --assume-extension --enforce_https --check_internal_hash --check_external_hash --ignore_empty_alt --log_level=debug --ignore_urls \"${URL_IGNORE}\" ./public"
-	#${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --swap_attributes ${CONFIG} ./public"
+	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume_extension --allow_hash_href true --allow_missing_href true ignore_empty_alt true --ignore_missing_alt true --check-external-hash false --check_internal_hash false --enforce_https false --ignore_urls \"${URL_IGNORE}\""
 

--- a/Makefile
+++ b/Makefile
@@ -69,7 +69,10 @@ NEW_URLS=$(shell scripts/ignore_new_urls.sh 2> /dev/null)
 URL_IGNORE:=$(URL_IGNORE)$(NEW_URLS)
 
 ## validate-site: Builds the site and validates the pages. This is used for CI
+CONFIG='{\"enforce_https\": false,\"check_internal_hash\": false, \"check_external_hash\":false,\"ignore_empty_alt\":true}'
 .PHONY: validate-site
 validate-site: build-hugo
-	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --assume-extension --check-external-hash --empty_alt_ignore --url-ignore \"${URL_IGNORE}\" ./public"
+	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --assume-extension --enforce_https --check_internal_hash --check_external_hash --ignore_empty_alt --ignore_urls \"${URL_IGNORE}\" ./public"
+	#${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --assume-extension --enforce_https --check_internal_hash --check_external_hash --ignore_empty_alt --log_level=debug --ignore_urls \"${URL_IGNORE}\" ./public"
+	#${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer --swap_attributes ${CONFIG} ./public"
 

--- a/Makefile
+++ b/Makefile
@@ -71,5 +71,5 @@ URL_IGNORE:=$(URL_IGNORE)$(NEW_URLS)
 ## validate-site: Builds the site and validates the pages. This is used for CI
 .PHONY: validate-site
 validate-site: build-hugo
-	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume_extension --allow_hash_href true --allow_missing_href true ignore_empty_alt true --ignore_missing_alt true --check_external_hash false --check_internal_hash false --enforce_https false --ignore_urls \"${URL_IGNORE}\""
+	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume_extension --allow_hash_href true --allow_missing_href true --ignore_empty_alt true --ignore_missing_alt true --check_external_hash false --check_internal_hash false --enforce_https false --ignore_urls \"${URL_IGNORE}\""
 

--- a/Makefile
+++ b/Makefile
@@ -69,7 +69,6 @@ NEW_URLS=$(shell scripts/ignore_new_urls.sh 2> /dev/null)
 URL_IGNORE:=$(URL_IGNORE)$(NEW_URLS)
 
 ## validate-site: Builds the site and validates the pages. This is used for CI
-CONFIG='{\"enforce_https\": false,\"check_internal_hash\": false, \"check_external_hash\":false,\"ignore_empty_alt\":true}'
 .PHONY: validate-site
 validate-site: build-hugo
 	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume_extension --allow_hash_href true --allow_missing_href true ignore_empty_alt true --ignore_missing_alt true --check_external_hash false --check_internal_hash false --enforce_https false --ignore_urls \"${URL_IGNORE}\""

--- a/Makefile
+++ b/Makefile
@@ -72,5 +72,5 @@ URL_IGNORE:=$(URL_IGNORE)$(NEW_URLS)
 CONFIG='{\"enforce_https\": false,\"check_internal_hash\": false, \"check_external_hash\":false,\"ignore_empty_alt\":true}'
 .PHONY: validate-site
 validate-site: build-hugo
-	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume_extension --allow_hash_href true --allow_missing_href true ignore_empty_alt true --ignore_missing_alt true --check-external-hash false --check_internal_hash false --enforce_https false --ignore_urls \"${URL_IGNORE}\""
+	${DORP} run -t -i --rm -v "$(shell pwd)":/site:z -w /site ${KIALI_HUGO_IMAGE} /bin/bash -c "npm prune && hugo && htmlproofer ./public --assume_extension --allow_hash_href true --allow_missing_href true ignore_empty_alt true --ignore_missing_alt true --check_external_hash false --check_internal_hash false --enforce_https false --ignore_urls \"${URL_IGNORE}\""
 

